flutter_local_notifications/test/ios_flutter_local_notifications_test.dart

```diff
@@ -760,5 +765,69 @@ void main() {
         isMethodCall('getNotificationAppLaunchDetails', arguments: null)
       ]);
     });
+
+    test('initialize with providesAppNotificationSettings', () async {
+      const DarwinInitializationSettings iosInitializationSettings =
+          DarwinInitializationSettings(
+        requestProvidesAppNotificationSettings: true,
+      );
+      const InitializationSettings initializationSettings =
+          InitializationSettings(iOS: iosInitializationSettings);
+      await flutterLocalNotificationsPlugin.initialize(initializationSettings);
+      expect(log, <Matcher>[
+        isMethodCall(
+          'initialize',
+          arguments: <String, Object>{
+            'requestAlertPermission': true,
+            'requestSoundPermission': true,
+            'requestBadgePermission': true,
+            'requestProvisionalPermission': false,
+            'requestCriticalPermission': false,
+            'requestProvidesAppNotificationSettings': true,
+            'defaultPresentAlert': true,
+            'defaultPresentSound': true,
+            'defaultPresentBadge': true,
+            'defaultPresentBanner': true,
+            'defaultPresentList': true,
+            'notificationCategories': <String>[],
+          },
+        ),
+      ]);
+    });
+
+    test('iOS requestPermissions with providesAppNotificationSettings',
+        () async {
+      await flutterLocalNotificationsPlugin
+          .resolvePlatformSpecificImplementation<
+              IOSFlutterLocalNotificationsPlugin>()!
+          .requestPermissions(
+              sound: true,
+              alert: true,
+              badge: true,
+              providesAppNotificationSettings: true);
+      expect(log, <Matcher>[
+        isMethodCall('requestPermissions', arguments: <String, Object>{
+          'sound': true,
+          'alert': true,
+          'badge': true,
+          'provisional': false,
+          'critical': false,
+          'providesAppNotificationSettings': true
+        })
+      ]);
+    });
+
+    test(
+      'check notification settings includes providesAppNotificationSettings',
```

Comment:
this test appears to be redundant as it has the same code as the `checkPermissions` test